app/controllers/migration_analytics_controller.rb

```diff
@@ -13,5 +13,30 @@ def layout_full_center
     end
   end
 
+  def payload
+    check_feature_enabled
+    task_id = params['task_id']
+    raise "Must specify a task id via \"task_id\"" if task_id.blank?
+
+    begin
+      task = MiqTask.find(task_id)
+      path = task&.context_data&.payload_path
+      if path && File.exist?(path)
+        send_file(path, :type => 'application/binary', :filename => 'analytics_payload.bin')
```

Comment:
Is it possible to parse out the filename as it is on disk and retain it instead of using `analytics_payload.bin`?